### PR TITLE
patvars

### PR DESCRIPTION
**Pull request by mcovarr**
Simple pattern variables compiler flag cleanup.

### CODE REVIEW

**Comment by geoffjentry** [comment]
🖖 👍 

I thought there were more than these but I guess not. Didn't have the energy to go back and do them again once I abandoned my first branch :)

**Comment by mcovarr** [comment]
Nah, this was pretty straightforward if a bit tedious.  `-Ywarn-unused:params` is a genuine beast, but some of the things that points out definitely look fishy.

**Comment by geoffjentry** [comment]
Oh yeah. IIRC the irritation I had here was that it'd not give me all the errors at once so i'd think i was done only to get hit by yet another block :)

I mentioned the unused vars in slack. There are some *definite* unused ones in there. Most of them were due to our (over?)reliance on overloaded function signatures where we may or may not use all of the args. Some of them were due to our pattern (particularly in the backend) of supplying a default no-op implementation of things. 

There were a handful though that were really bizarre. As in I couldn't figure out why the compiler thought it was unused.

**Comment by geoffjentry** [comment]
BTW you can see a lot of my crimes against humanity here: https://github.com/broadinstitute/cromwell/compare/jg_scala_2_12?expand=1

I had a variety of terrible ways of "using" the unused args. blech.

File: project/Settings.scala [hunk 1]
```diff
@@ -68,7 +68,8 @@ object Settings {
     "-Ywarn-unused:imports",
     "-Ywarn-unused:privates",
     "-Ywarn-unused:locals",
-    "-Xfatal-warnings"
+    "-Xfatal-warnings",
+    "-Ywarn-unused:patvars"
```
Comment (kshakir) [inline project/Settings.scala]: Move this up so that it's grouped with the other `-Yunsused:…`.

Otherwise 👍

**Comment by kshakir** [comment]
One minor comment. Otherwise 👍

**Pull request merged**